### ISSUE TITLE
unexpected match failure

### ISSUE THREAD

**Issue by andrewchambers** [janet-lang/janet#297]
```
> (match [123] [v] true)
true
> (match [nil] [v] true)
nil
```

**Comment by andrewchambers** [comment]
Simplified:

```
> (match nil v true)
nil
```

**Comment by bakpakin** [comment]
Fixed in 47e8f669f5bc5f5c10d93ad2ebaf71690c98356d

**Comment by andrewchambers** [comment]
thanks this fixes a bug in my postgres transaction code.
